src/Core/SemanticVersion.cs

```diff
@@ -39,32 +45,49 @@ public SemanticVersion(int major, int minor, int build, string specialVersion)
         {
         }
 
+        public SemanticVersion(int major, int minor, int build, string specialVersion, string metadata)

```

Comment:
I agree, but changing any of these would be a breaking change. The goal is here to allow the current code to read and sort SemVer 2.0.0 so that other projects can upgrade without making any other changes.

SpecialVersion is especially confusing, this class still treats versions as System.Version + a special pre-release label string since NuGet originally did not have pre-release.

I have made all of these changes in the past so that NuGet.Core.dll used a completely rewritten versioning class like NuGet.Version has. It took 1+ week of just fixing everywhere that SematicVersion was used along with the mock tests. The result was a change so large that it was never taken. This was all done ~2 years ago.